### DISCUSSION TITLE
Display HISTCMD in prompt?

### DISCUSSION THREAD

**Discussion by speedenator** [starship/starship#6058] [Q&A]
As the title says... I'd like to show $HISTCMD (the # of the history command) in the prompt. This is done via %h in a ZSH Powerlevel10k theme, or just using the (non-environment) variable $HISTCMD.

Couldn't see any module with this, and sadly custom via shell launches a NEW shell, where of course something like "echo $HISTCMD" returns 0 as it's the first command in a new shell.... (if there's a way to run a command in the current shell, that'd likely be all I need!)

Thanks!

**Comment by ethereal-faith** [comment]
```
[custom.command_number]
command = "echo %!"
when = true
format = "[$output ](bold purple)"
```

and add 'custom' to 'format' like 
format = """
$all\
$fill\
$cmd_duration\
$time\
$line_break\
$custom\
$character
"""

**Reply by ethereal-faith** [reply]
But it is broken in the latest version 1.20.1, I don't know why, but it works fine in version 1.19

**Reply by davidkna** [reply]
Please try this:
```toml
[custom.command_number]
when = true
format = "[%! ](bold purple)"
```

Command output is now escaped unless `unsafe_no_escape` is enabled.

**Reply by ethereal-faith** [reply]
I tried, but it still doesn't work

**Comment by speedenator** [comment]
in case anyone ever gets here, I figured it out. Here's what I use:

[custom.hist]
unsafe_no_escape = true
command = 'echo %!'
style = "fg:#b58900"
when = true


and in format:
format = """\
...
${custom.hist}\
...
"""

**Reply by brunogodbout** [reply]
Hi @speedenator .  I was excited to read your solution.  Unfortunately it did not work for me.  I get `%!` instead of the the HISTCMD value.  I am using bash.
